### PR TITLE
Fixed an issue where files in test dirs where being validated

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/28447

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -852,34 +853,32 @@ def filter_changed_files(self, files_string, tag='master', print_ignored_files=F
                     not (file_path.endswith('_test.py') or file_path.endswith('.Tests.ps1')):
                 # naming convention - code file and yml file in packages must have same name.
                 file_path = os.path.splitext(file_path)[0] + '.yml'
-
             # ignore changes in JS files and unit test files.
             elif file_path.endswith('.js') or file_path.endswith('.py') or file_path.endswith('.ps1'):
                 self.ignored_files.add(file_path)
                 continue
+            # ignore changes in TESTS_DIRECTORIES files.
+            elif any(file_path in test_dir for test_dir in TESTS_DIRECTORIES):
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:861]:
```suggestion
            elif any(test_dir in file-path for test_dir in TESTS_DIRECTORIES):
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -852,34 +853,32 @@ def filter_changed_files(self, files_string, tag='master', print_ignored_files=F
...
+                self.ignored_files.add(file_path)
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:862]:
```suggestion
                self.ignored_files.add(file_path)
                if print_ignored_files:
                     click.secho('Ignoring file path: {}'.format(file_path), fg="yellow")
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -852,34 +853,32 @@ def filter_changed_files(self, files_string, tag='master', print_ignored_files=F
                     not (file_path.endswith('_test.py') or file_path.endswith('.Tests.ps1')):
                 # naming convention - code file and yml file in packages must have same name.
                 file_path = os.path.splitext(file_path)[0] + '.yml'
-
             # ignore changes in JS files and unit test files.
             elif file_path.endswith('.js') or file_path.endswith('.py') or file_path.endswith('.ps1'):
                 self.ignored_files.add(file_path)
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:858]: if you are here - we might want to add this here as well:
```
if print_ignored_files:
        click.secho('Ignoring file path: {}'.format(file_path), fg="yellow")
```

**Review by roysagi — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/validate/validate_manager.py (hunk 3 above)
Comment (bakatzir) [reply demisto_sdk/commands/validate/validate_manager.py:858]: Added in a few other places as well.

**Comment by bakatzir** [comment]
```
Process finished with exit code 0
Setups class
PASSED     [100%]Ignoring file path: /Users/bkatzir/dev/demisto/demisto-sdk/demisto_sdk/tests/test_files/.pack-ignore
Ignoring file path: /Users/bkatzir/dev/demisto/demisto-sdk/demisto_sdk/tests/test_files/.secrets-ignore
Ignoring file path: /Users/bkatzir/dev/demisto/demisto-sdk/demisto_sdk/tests/test_files/integration_test.py
Ignoring file path: /Users/bkatzir/dev/demisto/demisto-sdk/demisto_sdk/tests/test_files/fake_pack/Integrations/test_data/results.json
Tearing down class
```

**Review by roysagi — APPROVED** [review]

**Pull request merged**